contracts/src/kleros-v1/KlerosV1Governor.sol

```diff
@@ -0,0 +1,190 @@
+        require(klerosLiquidDisputeIDtoGatewayDisputeID[_disputeID] == 0, "Dispute already relayed");
+        IKlerosLiquid.Dispute memory KlerosLiquidDispute = klerosLiquid.disputes(_disputeID);
+        (uint256[] memory votesLengths, , uint256[] memory totalFeesForJurors, , , ) = klerosLiquid.getDispute(
+            _disputeID
+        );
+
+        // Check that no juror was yet drawn. Add a function to finalize the juror drawing and move the dispute to vote.
+        require(KlerosLiquidDispute.period == IKlerosLiquid.Period.evidence, "Invalid dispute period.");
+        require(votesLengths.length == 1, "Cannot relay appeals.");
+
+        klerosLiquid.executeGovernorProposal(address(this), totalFeesForJurors[0], "");
+
+        uint256 minJurors = votesLengths[0];
+        bytes memory extraData = abi.encode(KlerosLiquidDispute.subcourtID, minJurors);
+        uint256 arbitrationCost = foreignGateway.arbitrationCost(extraData);
+        require(totalFeesForJurors[0] >= arbitrationCost, "Fees not high enough."); // If this doesn't hold at some point, it could be a big issue.
+        uint256 gatewayDisputeID = foreignGateway.createDispute(KlerosLiquidDispute.numberOfChoices, extraData);
+        klerosLiquidDisputeIDtoGatewayDisputeID[_disputeID] = gatewayDisputeID;
+        require(gatewayDisputeID != 0, "ID must be greater than 0.");
+
+        DisputeData storage dispute = disputes[gatewayDisputeID];
+        dispute.klerosLiquidDisputeID = _disputeID;
+    }
+
+    /** @dev Give a ruling for a dispute. Can only be called by the arbitrator. TRUSTED.
+     *  Triggers rule() from KlerosLiquid to the arbitrable contract which created the dispute.
+     *  @param _disputeID ID of the dispute in the arbitrator contract.
+     *  @param _ruling Ruling given by the arbitrator. Note that 0 is reserved for "Refused to arbitrate".
+     */
+    function rule(uint256 _disputeID, uint256 _ruling) public {
+        require(msg.sender == address(foreignGateway), "Not the arbitrator.");
+        DisputeData storage dispute = disputes[_disputeID];
+        require(dispute.klerosLiquidDisputeID != 0, "Dispute does not exist.");
+        require(!dispute.ruled, "Dispute already ruled.");
+
+        dispute.ruled = true;
+
+        emit Ruling(foreignGateway, _disputeID, _ruling);
+
+        IKlerosLiquid.Dispute memory klerosLiquidDispute = klerosLiquid.disputes(dispute.klerosLiquidDisputeID);
+
+        bytes4 functionSelector = IArbitrable.rule.selector;
+        bytes memory data = abi.encodeWithSelector(functionSelector, dispute.klerosLiquidDisputeID, _ruling);
+        klerosLiquid.executeGovernorProposal(klerosLiquidDispute.arbitrated, 0, data);
+    }
+
+    /** @dev Registers jurors' tokens which where locked due to relaying a given dispute. These tokens don't count as locked.
+     *  @param _disputeID The ID of the dispute as defined in KlerosLiquid.
+     */
+    function notifyFrozenTokens(uint256 _disputeID) external {
```

Comment:
When should this function be called?

If after a dispute is successfully relayed then probably we can call it from the `relayDispute` function itself.

Reply:
I have doubts about this part too, currently discussing it with Fede.

Reply:
It should be called:
 if tokensAtStakePerJuror > 0 && jurors were drawn && all jurors were drawn
 
 Note that jurors could get drawn after the dispute is relayed.